internal/npm_install/pre_process_package_json.js

```diff
@@ -53,9 +53,13 @@ function main() {
     // Note: there is no equivalent npm functionality to clean out individual packages
     // from the npm cache.
     clearYarnFilePathCaches(pkg);
+    console.log(JSON.stringify({pnp: isPnp(pkg)}));
```

Comment:
```suggestion
    process.stdout.write(JSON.stringify({pnp: isPnp(pkg)}));
```
makes it more obvious that this is a deliberate print to sdtout.